### DISCUSSION TITLE
Frage zum lexer und parser

### DISCUSSION THREAD

**Discussion by Thepigcat76** [BenediktLamberty/EigeneProgrammiersprache#7] [General]
Verwandelt dein Lexer den source code erst vollständig in tokens oder arbeitet er direkt hand in hand mit dem parser?

**Comment by BenediktLamberty** [comment]
Der Lexer teilt den Code erst vollständig in Tokens ein. Für manche Zwecke ist es in der Tat besser, wenn der Parser dem Lexer sagen könnte, was als nächstes erwartet wird oder umgekehrt. Allerdings brauche ich das für meine Zwecke nicht. 

Es freut mich, dass du weiterhin interessiert bis :)

**Reply by Thepigcat76** [reply]
Ah, vielen Dank!
